pkg/kubelet/volumemanager/populator/desired_state_of_world_populator.go

```diff
@@ -505,19 +505,21 @@ func (dswp *desiredStateOfWorldPopulator) createVolumeSpec(
 			pvcSource.ClaimName,
 			pvcUID)
 
-		// TODO: remove feature gate check after no longer needed
+		// TODO: remove features.BlockVolume checks / comments after no longer needed
+		volumeMode, err := util.GetVolumeMode(volumeSpec)
```

Comment:
When feature gate is off, then VolumeMode field will be nil.  Then this function will return an error.

Reply:
why didn't any pull jobs (unit/e2e tests) catch this? We should fix that

Reply:
> why didn't any pull jobs (unit/e2e tests) catch this? We should fix that

The test is marked as [Slow]. 

https://github.com/kubernetes/kubernetes/blob/1bb7835f0aef486c9037cd83d0e59f7ee48c5551/test/e2e/storage/testsuites/volumemode.go#L287

Reply:
Fixed. Good catch!

Reply:
>> why didn't any pull jobs (unit/e2e tests) catch this? We should fix that

>The test is marked as [Slow].

It's more complicated... We run Slow tests in e2e. The reason it passes is that API server defaults volumeMode to Filesystem:

https://github.com/kubernetes/kubernetes/blob/5e8ccda71cc676e070cd0f5d7b6990b8ebc24c1f/pkg/apis/core/v1/defaults.go#L279-L282

(btw, it does so even when the BlockVolume feature gate is off).

I am not sure it's actually possible to get `volumeMode = nil`.

Reply:
Rest strategy [drops fields](https://github.com/kubernetes/kubernetes/blob/5e8ccda71cc676e070cd0f5d7b6990b8ebc24c1f/pkg/api/persistentvolume/util.go#L27:6) when feature gate is off. Unsure if this comes into play when reading from a k8s client.